azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/handlers/v2/ArtifactHandlerV2.java

```diff
@@ -0,0 +1,135 @@
+            if (isMixingWarArtifactWithOtherArtifacts(allArtifacts)) {
+                mojo.getLog().warn(
+                    "Deploying war artifact together with other kinds of artifacts is not suggested," +
+                        " it will cause the content be overwritten or path incorrect issues.");
+            }
+            publishArtifactsViaZipDeploy(target, stagingDirectoryPath);
+        }
+    }
+
+    protected void publishArtifactsViaZipDeploy(final DeployTarget target,
+                                                final String stagingDirectoryPath) throws MojoExecutionException {
+        final File stagingDirectory = new File(stagingDirectoryPath);
+        final File zipFile = new File(stagingDirectoryPath + ".zip");
+        ZipUtil.pack(stagingDirectory, zipFile);
+        mojo.getLog().info(String.format("Deploying the zip package %s...", zipFile.getName()));
+
+        // Add retry logic here to avoid Kudu's socket timeout issue.
+        // More details: https://github.com/Microsoft/azure-maven-plugins/issues/339
+        int retryCount = 0;
+        while (retryCount < DEFAULT_MAX_RETRY_TIMES) {
+            retryCount += 1;
+            try {
+                target.zipDeploy(zipFile);
+                return;
+            } catch (Exception e) {
+                mojo.getLog().debug(
+                    String.format("Exception occurred when deploying the zip package: %s, " +
+                        "retrying immediately (%d/%d)", e.getMessage(), retryCount, DEFAULT_MAX_RETRY_TIMES));
+            }
+        }
+
+        throw new MojoExecutionException(String.format("The zip deploy failed after %d times of retry.", retryCount));
+    }
+
+    protected void publishArtifactsViaWarDeploy(final DeployTarget target, final String stagingDirectoryPath,
+                                                final List<File> warArtifacts) throws MojoExecutionException {
+        if (warArtifacts == null || warArtifacts.size() == 0) {
+            throw new MojoExecutionException(
+                String.format("There is no war artifacts to deploy in staging path %s.", stagingDirectoryPath));
+        }
+        for (final File warArtifact : warArtifacts) {
+            final String contextPath = getContextPathFromFileName(stagingDirectoryPath, warArtifact.getAbsolutePath());
+            publishWarArtifact(target, warArtifact, contextPath);
+        }
+    }
+
+    public void publishWarArtifact(final DeployTarget target, final File warArtifact,
+                                   final String contextPath) throws MojoExecutionException {
+        final Runnable executor = getRealWarDeployExecutor(target, warArtifact, contextPath);
+        mojo.getLog().info(String.format("Deploying the war file %s...", warArtifact.getName()));
+
+        // Add retry logic here to avoid Kudu's socket timeout issue.
+        // More details: https://github.com/Microsoft/azure-maven-plugins/issues/339
+        int retryCount = 0;
```

Comment:
has some duplicated code here. Is it possible to extract the logic into a util function?